internal/appsec/appsec.go

```diff
@@ -135,3 +132,27 @@ func (a *appsec) stop() {
 		a.limiter.Stop()
 	}
 }
+
+func (a *appsec) startRC() {
+	if a.rc == nil {
+		return
+	}
+	// Set WAF-related RC capabilities and products if the WAF is in good health. We perform this check in order not to
+	// falsely "allow" users to activate ASM through remote config if activation would fail when trying to register a WAF handle
+	// (ex: if the service runs on an unsupported platform).
+	if err := waf.Health(); err == nil {
+		a.rc.Capabilities = append(a.rc.Capabilities, remoteconfig.ASMActivation)
```

Comment:
nit: Can we flip this `if` here to align the happy path to the left? 
Just have a log.Debug and a `return` here perhaps?

Reply:
Done. Added a defer for `a.rc.Start()` as well since we want to start the RC client even if `waf.Health()` returns an error